### DISCUSSION TITLE
Command line help with .bat file

### DISCUSSION THREAD

**Discussion by corbrink** [ImageMagick/ImageMagick#2941] [Consulting]
I need to resize pictures without cropping them but do not know the command line tools.

This is what I have:
for %%f in (\photos\*.jpg) do magick convert -size 600x600 xc:white %%f -gravity center -composite padded\%%f
mkdir \photos\padded\

The bat runs but there is no output or converted files. I obviously need to save the new files in padded but do not know how.
Any help will be greatly appreciated.

**Comment by snibgo** [comment]
Do you have a directory called "padded"? Why are you creating a directory _after_ converting all the files? Do you have any files in root named "photos*.jpg"?

For testing, I suggest you replace the IM command with "echo %%f".

**Reply by corbrink** [reply]
yes and yes

**Reply by corbrink** [reply]
I do not need to create padded

**Comment by fmw42** [comment]
Is the directory padded already created? ImageMagick will not create directories. If padded is not in your current working directory, then you must provide the path to that directory.

**Reply by corbrink** [reply]
yes it is

**Reply by corbrink** [reply]
You still don't tell me how to save the converted files to padded

**Comment by corbrink** [comment]
Not very helpful. I want to know the commands to convert & save the converted files to a new directory please.

**Comment by snibgo** [comment]
What happens when you use this command, in a BAT script:
```
for %%f in (\photos*.jpg) do echo %%f
```
If nothing is echoed, then no input files were found, so the magick command would never be executed.

Also try:
```
dir \photos*.jpg
```

**Comment by russbroom** [comment]
Shouldn't `"\photos*.jpg"` be `"\photos\*.jpg"`?
